### PR TITLE
Implement EncryptionFeatures

### PR DESCRIPTION
**Pull request by RohrerF**
- Adds commands to encrypt and decrypt files
  - `confix encrypt secrets.json secrets.enc`
  - `confix decrypt secrets.enc secrets.json`
  - `confix buid --encrypt`
- Implemented Providers
  - Azure KeyVault 

#87

### CODE REVIEW

File: src/Confix.Tool/src/Confix.Tool/Middlewares/Encryption/Providers/AzureKeyVault/AzureKeyVaultEncryptionProvider.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    public AzureKeyVaultEncryptionProvider(CryptographyClient client)
+    {
+        _client = client;
+    }
+
+    public const string Type = "AzureKeyVault";
+
+    public Task<byte[]> DecryptAsync(byte[] data, CancellationToken cancellationToken)
+    => KeyVaultExtension.HandleKeyVaultException(async () =>
```
Comment (PascalSenn) [inline src/Confix.Tool/src/Confix.Tool/Middlewares/Encryption/Providers/AzureKeyVault/AzureKeyVaultEncryptionProvider.cs:34]: I guess at this point we should just write a decorated keyvault client. Lets do it it a follow up

File: src/Confix.Tool/src/Confix.Tool/Middlewares/Encryption/Providers/AzureKeyVault/AzureKeyVaultEncryptionProviderConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+using System.Text.Json.Nodes;
+using ConfiX.Extensions;
+
+namespace Confix.Tool.Middlewares.Encryption.Providers.AzureKeyvault;
+
+public sealed record AzureKeyVaultEncryptionProviderConfiguration(
+    string? Uri,
+    string? KeyName,
+    string? KeyVersion
+)
```
Comment (PascalSenn) [inline src/Confix.Tool/src/Confix.Tool/Middlewares/Encryption/Providers/AzureKeyVault/AzureKeyVaultEncryptionProviderConfiguration.cs:11]: this is not how .net formats things btw. Even if there is no prettier, the formatting should be consistent.

File: src/Confix.Tool/src/Confix.Tool/Pipelines/Encryption/OutputFileArgument.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+internal sealed class OutputFileArgument : Argument<FileInfo?>
+{
+    public static OutputFileArgument Instance { get; } = new();
+
+    private OutputFileArgument()
+        : base("out-file")
+    {
+        Arity = ArgumentArity.ZeroOrOne;
+        Description = """
+            The file to write the encrypted or decrypted data to.
```
Comment (PascalSenn) [inline src/Confix.Tool/src/Confix.Tool/Pipelines/Encryption/OutputFileArgument.cs:14]: How does that look in the output?

**Review by PascalSenn — APPROVED** [review]

**Comment by PascalSenn** [comment]
some small things that we can resolve later. Lets get the MVP out

File: src/Confix.Tool/src/Confix.Tool/Pipelines/Encryption/OutputFileArgument.cs (hunk 3 above)
Comment (RohrerF) [reply src/Confix.Tool/src/Confix.Tool/Pipelines/Encryption/OutputFileArgument.cs:14]: image

**Pull request merged**